File: clinical-mdr-api/clinical_mdr_api/domain/study_selection/study_activity_schedule.py
import datetime
from dataclasses import dataclass
from typing import Optional


@dataclass
class StudyActivityScheduleVO:
    study_uid: str
    study_activity_uid: Optional[str]
    study_activity_name: Optional[str]
    study_visit_uid: Optional[str]
    study_visit_name: Optional[str]
    note: Optional[str]

    # Study selection Versioning
    start_date: datetime.datetime
    user_initials: Optional[str]

    uid: Optional[str] = None

File: studybuilder-ddf-api-adaptor/ddf-translator-lib/src/main/java/org/openstudybuilder/model/StudyActivitySchedule.java
package org.openstudybuilder.model;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.Getter;
import lombok.Setter;

/**
 * Maps to StudyActivitySchedule
 */
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@Getter @Setter
public class StudyActivitySchedule {
    private String studyUid;
    private String studyActivityScheduleUid;
    private String studyActivityUid;
    private String studyActivityName;
    private String studyVisitUid;
    private String studyVisitName;
    private String note;
    private String startDate;
    private String endDate;
    private String userInitials;
}
